Source/Virtusize.swift

```diff
@@ -230,6 +209,45 @@ public class Virtusize {
         return apiResponse
     }
 
+	internal class func updateSession() {
+		let userSessionInfoResponse = Virtusize.getUserSessionInfoAsync()
+		if let accessToken = userSessionInfoResponse.success?.id {
+			UserDefaultsHelper.current.accessToken = accessToken
+		}
+		if let authHeader = userSessionInfoResponse.success?.authHeader, !authHeader.isEmpty {
+			UserDefaultsHelper.current.authHeader = authHeader
+		}
+	}
+
+	internal class func setupRecommendation(selectedUserProductId: Int? = nil, loggedOutUser: Bool = false) {
+		var userProducts: [VirtusizeInternalProduct]?
+		var userBodyProfile: VirtusizeUserBodyProfile?
+		bodyProfileRecommendedSize = nil
+		sizeComparisonRecommendedSize = nil
+		if !loggedOutUser {
+			userProducts = Virtusize.getUserProductsAsync().success
+			userBodyProfile = Virtusize.getUserBodyProfileAsync().success
+		}
+		if userProducts != nil && productTypes != nil && storeProduct != nil && userBodyProfile != nil {
```

Comment:
You should rather use the `let` conditional assignment here, so you don't have to force cast the type later

```swift
if let userProducts = userProducts,... { ... }
```

Reply:
Done